### PR TITLE
Moves the Soap Test suite to Pest

### PR DESCRIPTION
**Pull request by lukeraymonddowning**

### CODE REVIEW

**Comment by lukeraymonddowning** [comment]
Want to hold on this until https://github.com/pestphp/pest/pull/331 because a lot of this can be refactored to use that.

File: tests/Unit/BasicSyntaxTest.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                'DivideResponse Divide(Divide $parameters)',
+            ]
+        );
+
+    expect(soap(null, $mock)->to(EXAMPLE_SOAP_ENDPOINT)->functions())
+        ->toBeArray()
+        ->not->toBeEmpty();
+});
+
+it('is empty', function() {});
```
Comment (nuernbergerA) [inline tests/Unit/BasicSyntaxTest.php:28]: seams like dead code or?

Comment (lukeraymonddowning) [reply tests/Unit/BasicSyntaxTest.php]: Thanks!

File: tests/Unit/BasicSyntaxTest.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+use Mockery as m;
+use RicorocksDigitalAgency\Soap\Request\Request;
+use RicorocksDigitalAgency\Soap\Response\Response;
+
+it('can obtain a WSDL', function () {
+    $mock = m::mock(Request::class);
+    $mock->shouldReceive('beforeRequesting', 'afterRequesting', 'to')->andReturnSelf()
+        ->shouldReceive('functions')->andReturn(
+            [
+                'AddResponse Add(Add $parameters)',
```
Comment (nedwors) [inline tests/Unit/BasicSyntaxTest.php:12]: We've set up the mock return value but then not asserted against it. Could we do that? And with it being a mock, could we reduce the number of items in the array?

File: tests/Unit/BasicSyntaxTest.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        );
+
+    expect(soap(null, $mock)->to(EXAMPLE_SOAP_ENDPOINT)->functions())
+        ->toBeArray()
+        ->not->toBeEmpty();
+});
+
+it('can call a SOAP function')
+    ->fake(['*' => Response::new(['AddResult' => 35])])
+    ->expect(fn () => $this->soap()->to(EXAMPLE_SOAP_ENDPOINT)->Add(['intA' => 10, 'intB' => 25])->AddResult)
```
Comment (nedwors) [inline tests/Unit/BasicSyntaxTest.php:30]: I think for this test this should be `call('Add', ...)`

**Review by nedwors — APPROVED** [review]
Looking really sweet @lukeraymonddowning 🔥 

I feel that some of the more complex HOTs _could_ benefit from becoming standard tests, just due to their verbosity - thinking the back end of `SoapFakeTest`, `MethodFakeTest`. But it's really subjective and I completely understand that to break out of HOT in a HOT test file is a bit, well, bleugh.

Approved pending the comments.

File: tests/Unit/BasicSyntaxTest.php (hunk 2 above)
Comment (lukeraymonddowning) [reply tests/Unit/BasicSyntaxTest.php:12]: Updated

File: tests/Unit/BasicSyntaxTest.php (hunk 3 above)
Comment (lukeraymonddowning) [reply tests/Unit/BasicSyntaxTest.php]: Updated

**Pull request merged**